### PR TITLE
WIP: replace methadone and trollop with GLI

### PR DESCRIPTION
**Pull request by aussielunix**
Thanks for writing flapjack-syrup !
One thing I had some trouble with when looking at the code was how to extend this.
There is a combination of methadone (not suitable to command suites) and trollop and I was a little confused about where to declare the UI for any new features.

Not sure if you have seen [gli](http://naildrivin5.com/gli/) (same author as methadone) before but this is a great rubygem that lets you separate the UI from the code and gives you a sweet dsl for expressing that UI of your application.

I have mapped out the UI using gli.  I am not quite finished yet but if this is somehting you like and willing to merge in I will finish this off in the next few days.

### CODE REVIEW

**Comment by aussielunix** [comment]
Why I started this PR:

By having a clear separation of UI and app logic it becomes:
- easily readable code layout / very easy to contribute code in the future
- easy to test the UI with cucumber/aruba
- easy to do outside in testing/development
  - write cucumber/aruba feature testing the UI
  - add new CLI feature (using GLI's dsl)
  - if tests pass then write the app logic for the new feature
- auto formatted `--help` populated from the easy GLI dsl
- [shell completion](https://github.com/davetron5000/gli/wiki/ShellCompletion) of commands and subcommands etc
- Defaults for options/args set in the DSL plus ability to allow users to set their own defaults via [config](https://github.com/davetron5000/gli/wiki/Config) file.

P.S. GLI is awesome

**Comment by ghicks-rmn** [comment]
Sorry for the delay in responding, I was out for the holidays. I'm totally open to moving to GLI.

I realize that using methadone plus trollop is a little odd - I started with methadone to try and get the gem infrastructure quickly and easily, and then found its limitations when I decided that a command-based structure would be easiest to work with.  It sounds like GLI is the better way to go.

File: lib/flapjack/syrup/ui/medium.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+medium crud - bunch of sub commands for medium manipulation
+'''
+command :medium do |c|
+  c.desc 'create a new medium'
+  c.command :create do |create|
+    create.action do |global_options,options,args|
+      puts "TODO: medium create"
+    end
+  end
+  c.desc 'get an madium'
```
Comment (rickatnight11) [inline lib/flapjack/syrup/ui/medium.rb:12]: `get details on a medium`

File: lib/flapjack/syrup/ui/medium.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  c.command :get do |get|
+    get.action do |global_options,options,args|
+      puts "TODO: medium get"
+    end
+  end
+  c.desc 'update an mediu'
```
Comment (rickatnight11) [inline lib/flapjack/syrup/ui/medium.rb:18]: `update a medium`

File: lib/flapjack/syrup/ui/medium.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  c.command :update do |update|
+    update.action do |global_options,options,args|
+      puts "TODO: medium update"
+    end
+  end
+  c.desc 'delete an medium'
```
Comment (rickatnight11) [inline lib/flapjack/syrup/ui/medium.rb:24]: `delete a medium`

**Pull request closed without merging**